include/bitcoin/bitcoin/utility/random.hpp

```diff
@@ -19,40 +19,52 @@
 #ifndef LIBBITCOIN_RANDOM_HPP
 #define LIBBITCOIN_RANDOM_HPP
 
+#include <random>
 #include <cstdint>
 #include <bitcoin/bitcoin/define.hpp>
 #include <bitcoin/bitcoin/utility/asio.hpp>
 #include <bitcoin/bitcoin/utility/data.hpp>
 
 namespace libbitcoin {
 
-/**
- * Generate a pseudo random number within the domain.
- * @return  The 64 bit number.
- */
-BC_API uint64_t pseudo_random();
+class random
+{
+  public:
+    /**
+     * Returns a thread-safe static random generator instance.
+     * @return  The generator.
+     */
+    static std::mt19937& get_twister();
```

Comment:
The idea of making random a class is to keep this and any other implementation details private while still being able to exposing them to templatized public members.

Reply:
What I like about how it is now is that the methods are exposed and can be templatized and any caller can now just replace *pseudo_random* with *random::pseudo_random* instead of creating a local instance and then calling that method on it.

Reply:
Since they were functions previously there was no instantiation. The only difference is in the predication of the class name. The v3 implementation can retain compatibility (eliminating the class name predication requirement) with a `using` declaration or a wrapper function.

Reply:
It is also worth reviewing the method names, given that they are collected in the scope of `random`. The word `random` in each function name is a redundancy that should be removed.